flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/nonha/standalone/StandaloneHaServices.java

```diff
@@ -116,7 +108,7 @@ public LeaderRetrievalService getJobManagerLeaderRetriever(JobID jobID) {
 		synchronized (lock) {
 			checkNotShutdown();
 
-			return new StandaloneLeaderRetrievalService(jobManagerAddress, DEFAULT_LEADER_ID);
+			return new StandaloneLeaderRetrievalService("UNKNOWN", DEFAULT_LEADER_ID);
```

Comment:
It seems `org.apache.flink.runtime.resourcemanager.JobLeaderIdService#addJob` still use this interface from outside. Is it affected?

Reply:
No it is not because it is only interested in the leader session id. The address is ignored.

Reply:
`HighAvailability#getJobManagerLeaderRetriever(JobID)` for which the address of the `StandaloneLeaderRetrievalService` is relevant will only be used by the `JobLeaderIdService` which is only interested in the leader session id and not the address. That's why we can change it.